.github/workflows/build_profiler.yml

```diff
@@ -1,42 +1,361 @@
-name: Build .NET Agent Profiler
+name: Build and Package Profiler
 
-# Controls when the workflow will run
 on:
-  # Triggers the workflow on push or pull request events but only for the "main" branch
-  #push:
-  #  branches: [ "main" ]
-  #pull_request:
-  #  branches: [ "main" ]
+  # run this workflow on a push to any branch other than main or a feature branch
+  push:
+    branches-ignore:
+      - main
+      - "feature/**"
 
-  # Allows you to run this workflow manually from the Actions tab
+  # this workflow can be called from another workflow
+  workflow_call:
+    inputs:
+      force-build:
+        description: 'Force a build, even if no files are changed'
+        required: true
+        type: boolean
+  
+  # this workflow can be invoked manually
   workflow_dispatch:
+    inputs:
+      force-build:
+        description: 'Force a build, even if no files are changed'
+        required: true
+        default: true
+        type: boolean
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 permissions:
   contents: read
 
+# only allow one instance of this workflow to be running per PR or branch, cancels any that are already running
+concurrency:
+  group: build-profiler-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  scripts_path: ${{ github.workspace }}\build\scripts
+  tools_path: ${{ github.workspace }}\build\Tools
+  DOTNET_NOLOGO: true
+
 jobs:
-  # This workflow contains a single job called "build"
-  build:
-    # The type of runner that the job will run on
-    runs-on: ubuntu-latest
+  check-for-changes:
+    name: Check for Updated Profiler Files
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: read
+    outputs:
+      profiler_src: ${{ steps.filter.outputs.profiler_src }}
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@03bee3930647ebbf994244c21ddbc0d4933aab4f # v2.3.0
```

Comment:
I think this needs the latest harden runner updates.

Reply:
The change got picked up when I merged from `main`.

Reply:
It looks like it didn't pick up the updates in this workflow. The All Solutions workflow is on 2.4.0, but this one is still on 2.3.0.